### PR TITLE
Release of API version 1.0.0-rc.1

### LINKED ISSUE #122: Scope of SimSwap for Fall24 CAMARA Release
This issue shall define which issues are within scope of APIs of the Sub Project for the Fall24 release and will be regularly updated based on the decision within the Sim Swap API meetings.
Note: this issue is only for sim_swap and not covering sim-swap-notification-subscription

Alignement with commonalities & icm
- PhoneNumber pattern - clarification on optional "+" prefix ([issue85](https://github.com/camaraproject/SimSwap/issues/85))
- Revisit defined errors ([issue119](https://github.com/camaraproject/SimSwap/issues/119))

Enhancement

- Add API-Name (aka wild-card) as  scope in the yaml ([issue103](https://github.com/camaraproject/SimSwap/pull/103))
- Provide test definition ([issue63](https://github.com/camaraproject/SimSwap/issues/63))
- Provide user stories([issue121](https://github.com/camaraproject/SimSwap/issues/121))

Fixes

- phone number instead of MSISDN to follow commonalities guidelines ([issue116](https://github.com/camaraproject/SimSwap/pull/116))
- Make response properties "latestSimChange" and "swapped" required since they will always be returned ([issue96](https://github.com/camaraproject/SimSwap/issues/96))

### PR DESCRIPTION
**Pull request by fernandopradocabrillo**
#### What type of PR is this?

Add one of the following kinds:
* documentation
* subproject management

#### What this PR does / why we need it:

Generation of first release-candidate for Sim Swap API

This first release candidate r1.1 contains the definition and documentation of the release-candidate of the Sim Swap API v1.0.0-rc.1.

#### Which issue(s) this PR fixes:

Fixes #122

### CODE REVIEW

**Review by bigludo7 — COMMENTED** [review]
Add the part for sim swap subscription

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,6 +88,36 @@
...
+    - [View it on ReDoc](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/SimSwap/blob/r1.1/code/API_definitions/sim-swap-subscriptions&nocors)
+    - [View it on Swagger Editor](https://editor.swagger.io/?url=https://raw.githubusercontent.com/camaraproject/SimSwap/blob/r1.1/code/API_definitions/sim-swap-subscriptions)
+
+**sim-swap-subscriptions v0.1.0alpha1 is the first alpha version for CAMARA Sim Swap subscription API**
+
+## Please note:
+
+- This is an **alpha version**, it should be considered as a **draft**
+- There are bug fixes to be expected and incompatible changes in upcoming versions 
+- The release is suitable for implementors, but it is not recommended to use the API with customers in productive environments
```
Comment (hdamker) [inline CHANGELOG.md:104]: Question: is this already the case with this version? Should implementors time already time on it? 

Within ... it says about alpha versions:

> The purpose of an alpha API version is to support rapid development of an API and the creation of test implementations for feedback purposes.

Proposal (the statement is about the API version, I suppose):

The API version is suitable for test implementations and has the purpose to collect feedback for its further development. It should not be used with customers in productive environments.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,8 +2,64 @@
...
+* Make response properties "latestSimChange" and "swapped" required since they will always be returned [Issue 96](https://github.com/camaraproject/SimSwap/issues/96)
+* Update SIM Swap & SIM Swap notification subscription yaml to make the '+' mandatory for phoneNumber by @bigludo7 in https://github.com/camaraproject/SimSwap/pull/100
+* Add README file to indicate shift to Confluence page by @bigludo7 in https://github.com/camaraproject/SimSwap/pull/101
+* phone number instead of MSISDN to follow communalities guidelines by @gregory1g in https://github.com/camaraproject/SimSwap/pull/116
+* Remove unused errors and align with commonalities errors by @fernandopradocabrillo in https://github.com/camaraproject/SimSwap/pull/126
+
+### Removed
+
+* n/a
+
```
Comment (hdamker) [inline CHANGELOG.md:55]: Here should follow the (new) section about "sim-swap-subscription v0.1.0-alpha.1" (it is part of r1.1).

The existing content about v0.4.0 must be kept unchanged.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,8 +2,64 @@
...
+## New Contributors
+
+* @trehman-gsma made their first contribution in https://github.com/camaraproject/SimSwap/pull/62
+* @gregory1g made their first contribution in https://github.com/camaraproject/SimSwap/pull/82
+* @rartych made their first contribution in https://github.com/camaraproject/SimSwap/pull/110
+* @AxelNennker made their first contribution in https://github.com/camaraproject/SimSwap/pull/103
+
```
Comment (hdamker) [inline CHANGELOG.md:62]:
```suggestion

**Full Changelog**: https://github.com/camaraproject/SimSwap/compare/v0.4.0...r1.1
```

**Review by hdamker — COMMENTED** [review]
@fernandopradocabrillo some suggestions ... the main point is that the v0.4.0 was rewritten. But it need to kept unchanged. The new section about sim-swap-subscriptions belongs into the r1.1 section

**Comment by fernandopradocabrillo** [comment]
Comments addressed. Thanks @hdamker!

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,8 +2,96 @@
...
+* Update SIM Swap & SIM Swap notification subscription yaml to make the '+' mandatory for phoneNumber by @bigludo7 in https://github.com/camaraproject/SimSwap/pull/100
+* Add README file to indicate shift to Confluence page by @bigludo7 in https://github.com/camaraproject/SimSwap/pull/101
+* phone number instead of MSISDN to follow communalities guidelines by @gregory1g in https://github.com/camaraproject/SimSwap/pull/116
+* Remove unused errors and align with commonalities errors by @fernandopradocabrillo in https://github.com/camaraproject/SimSwap/pull/126
+
+### Removed
+
+* n/a
+
+## Sim Swap Subscriptions v0.1.0-alpha1
```
Comment (hdamker) [inline CHANGELOG.md:55]:
```suggestion
## Sim Swap Subscriptions v0.1.0-alpha.1
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,8 +2,96 @@
...
+
+**sim-swap-subscriptions v0.1.0-alpha1 is the first alpha version for CAMARA Sim Swap subscription API**
```
Comment (hdamker) [inline CHANGELOG.md:57]:
```suggestion
**sim-swap-subscriptions v0.1.0-alpha.1 is the first alpha version for CAMARA Sim Swap subscription API**
```

File: documentation/API_documentation/API-Readiness-Checklist_SIM-Swap-1.0.0.md
Comment (hdamker) [inline documentation/API_documentation/API-Readiness-Checklist_SIM-Swap-1.0.0.md:1]: the filename shouldn't contain a version number and following the format, in result:
`sim-swap-API-Readiness-Checklist.md`

Line 3 should be then

Checklist for sim-swap 1.0.0-rc.1 in r1.1

File: README.md [hunk 6]
```diff
@@ -18,22 +18,26 @@ Repository to describe, develop, document and test the SimSwap API family
...
-## Contributorship and mailing list
-
-* To subscribe / unsubscribe to the mailing list of this Sub Project and thus be / resign as Contributor please visit <https://lists.camaraproject.org/g/sp-ssw>.
-* A message to all Contributors of this Sub Project can be sent using <sp-ssw@lists.camaraproject.org>.
+* **The latest pre-release of CAMARA Sim Swap API is 1.0.0-rc.1. This is the release candidate of the first stable version of the API**. It is suitable for implementors.
+* The Release Tag is [r1.1](https://github.com/camaraproject/SimSwap/releases/tag/r1.1).
+  - API definition **with inline documentation**:
+    - OpenAPI [YAML spec file](https://github.com/camaraproject/SimSwap/blob/r1.1/code/API_definitions/sim_swap.yaml)
+    - [View it on ReDoc](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/SimSwap/r1.1/code/API_definitions/sim_swap.yaml&nocors) 
+    - [View it on Swagger Editor](https://editor.swagger.io/?url=https://raw.githubusercontent.com/camaraproject/SimSwap/r1.1/code/API_definitions/sim_swap.yaml)
```
Comment (hdamker) [inline README.md:29]: This part does not fit completely, especially not as there a now two APIs in the release, with different versions.

A suggestion below, a slightly different format example you can find in https://github.com/camaraproject/DeviceStatus/pull/190

```suggestion
* The latest pre-release is [r1.1](https://github.com/camaraproject/SimSwap/releases/tag/r1.1). It contains:
  * **sim-swap v1.0.0-rc.1** 
    * **This is the release candidate of the first stable version of the Sim Swap API**. It is suitable for implementors.
    * API definitions (with inline documentation):
      * OpenAPI [YAML](https://github.com/camaraproject/SimSwap/blob/r1.1/code/API_definitions/sim_swap.yaml)
      * [View it on ReDoc](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/SimSwap/r1.1/code/API_definitions/sim_swap.yaml&nocors) 
      * [View it on Swagger Editor](https://editor.swagger.io/?url=https://raw.githubusercontent.com/camaraproject/SimSwap/r1.1/code/API_definitions/sim_swap.yaml)
  * sim-swap-swap-subscriptions v0.1.0-alpha.1
     * This is the first alpha version for CAMARA Sim Swap subscription API. It should considered as a draft.
     * API definitions (with inline documentation):
       * OpenAPI [YAML](https://github.com/camaraproject/SimSwap/blob/r1.1/code/API_definitions/sim-swap-subscriptions.yaml)
       * [View it on ReDoc](https://redocly.github.io/redoc/?url=https://raw.githubusercontent.com/camaraproject/SimSwap/r1.1/code/API_definitions/sim-swap-subscriptions.yaml&nocors) 
       * [View it on Swagger Editor](https://editor.swagger.io/?url=https://raw.githubusercontent.com/camaraproject/SimSwap/r1.1/code/API_definitions/sim-swap-subscriptions.yaml)
 ```

**Comment by hdamker** [comment]
> Comments addressed. Thanks @hdamker!

Thanks for the updates. I resolved the addressed and outdated conversations. 

And added then a few new ones, see above :-)

File: CHANGELOG.md [hunk 7]
```diff
@@ -2,8 +2,96 @@
...
+* for subsequent alpha or release-candidate API versions, the delta with respect to the previous pre-release
+* for a public API version, the consolidated changes since the release of the previous public API version
+
+# r1.1 - rc
+
+## Release Notes
+
+This release contains the definition and documentation of
+* Sim Swap API 1.0.0-rc.1
+* Sim Swap Subscriptions v0.1.0-alpha1
```
[the remark this answers is not in the document]
Comment (fernandopradocabrillo) [reply CHANGELOG.md:22]: yep sorry, I forgot the dot

**Comment by hdamker** [comment]
@fernandopradocabrillo I will do the final review "on behalf of release management" when you changed this from "draft" to "ready for review".

File: CHANGELOG.md [hunk 8]
```diff
@@ -2,8 +2,96 @@
...
+### Added
+
+* User Story in documentation/API_documentation directory by @jgarciahospital [PR125](https://github.com/camaraproject/SimSwap/pull/125)
+* Test Definition in Test_Definitions directory by @fernandopradocabrillo [PR70](https://github.com/camaraproject/SimSwap/pull/70)
+* add API-Name aka wild-card scope by @AxelNennker in https://github.com/camaraproject/SimSwap/pull/103
+
+### Changed
+* Make response properties "latestSimChange" and "swapped" required since they will always be returned [Issue 96](https://github.com/camaraproject/SimSwap/issues/96)
+* Update SIM Swap & SIM Swap notification subscription yaml to make the '+' mandatory for phoneNumber by @bigludo7 in https://github.com/camaraproject/SimSwap/pull/100
+* Add README file to indicate shift to Confluence page by @bigludo7 in https://github.com/camaraproject/SimSwap/pull/101
```
Comment (hdamker) [inline CHANGELOG.md:47]: This line is not really needed within the Changelog.

**Comment by hdamker** [comment]
https://github.com/camaraproject/SimSwap/blob/main/code/API_definitions/sim-swap-subscriptions.yaml:
x-camara-commonalities: 0.4.0 is missing in info object

**Comment by hdamker** [comment]
https://github.com/camaraproject/SimSwap/blob/main/code/API_definitions/sim_swap.yaml:
Authorization and Authentication part need to be updated, created #133 for that.

**Comment by hdamker** [comment]
@fernandopradocabrillo please find above the result from the review across files from release management perspective. If the comments are addressed (especially #133), the PR can get the approval from Release Management.

**Comment by fernandopradocabrillo** [comment]
> @fernandopradocabrillo please find above the result from the review across files from release management perspective. If the comments are addressed (especially #133), the PR can get the approval from Release Management.

1. Auth section updated
2. x-camara-commonalities included in subscription yaml
3. Unneded changelog registry removed

I think thats all! thanks @hdamker

**Review by hdamker — APPROVED** [review]
Looks good from Release Management perspective.

**Comment by hdamker** [comment]
@camaraproject/sim-swap_maintainers @AxelNennker Would be good if one of you will review and approve before I support @fernandopradocabrillo here with admin rights to merge the PR (@bigludo7 is on vacation).

**Comment by hdamker** [comment]
@fernandopradocabrillo there seem to be no additional comments. Should I merge the PR so that you can create the r1.1?

**Comment by fernandopradocabrillo** [comment]
> @fernandopradocabrillo there seem to be no additional comments. Should I merge the PR so that you can create the r1.1?

@hdamker Yes! thank you

**Comment by hdamker** [comment]
Merging on behalf Codeowners

**Pull request merged**